server/db/db.js

```diff
@@ -663,17 +663,37 @@ async function getInstallationWithSessionId(sessionId, clientParam) {
   return null
 }
 
-async function getHistoricAlertsByAlertApiKey(alertApiKey, maxHistoricAlerts, maxTimeAgoInMillis, clientParam) {
-  let client = clientParam
-  const transactionMode = typeof client !== 'undefined'
-
+async function getActiveAlertsByAlertApiKey(alertApiKey, maxTimeAgoInMillis, clientParam) {
   try {
-    if (!transactionMode) {
-      client = await pool.connect()
-    }
+    const results = await helpers.runQuery(
+      'getActiveAlertsByAlertApiKey',
+      `
+      SELECT s.id, s.state, b.unit, s.num_presses, i.incident_categories, s.created_at
+      FROM sessions AS s
+      LEFT JOIN buttons AS b ON s.button_id = b.button_id
+      LEFT JOIN installations AS i ON s.installation_id = i.id
+      WHERE i.alert_api_key = $1
+      AND (
+        s.state <> $2
+        AND s.updated_at >= now() - $3 * INTERVAL '1 millisecond'
+      )
+      ORDER BY s.created_at DESC
+      `,
+      [alertApiKey, CHATBOT_STATE.COMPLETED, maxTimeAgoInMillis],
+      pool,
+      clientParam,
+    )
 
-    // Historic Alerts are those with status "Completed" or that were last updated longer ago than the SESSION_RESET_TIMEOUT
-    const { rows } = await client.query(
+    return results.rows
+  } catch (err) {
+    helpers.logError(err.toString())
+  }
+}
+
+async function getHistoricAlertsByAlertApiKey(alertApiKey, maxHistoricAlerts, maxTimeAgoInMillis, clientParam) {
```

Comment:
While I was in the area, changed this one to use `helpers.runQuery` as well.

Reply:
nice, thanks!